Description: this patch adds `|| exit 1' to all grub-probe commands for
  which failure is fatal.

--- util/grub-install.in.orig	2010-01-02 14:42:38.000000000 +0100
+++ util/grub-install.in
@@ -259,7 +259,7 @@ for file in ${grubdir}/locale/*.mo ${pkg
 done
 
 # Write device to a variable so we don't have to traverse /dev every time.
-grub_device=`$grub_probe --target=device ${grubdir}`
+grub_device=`$grub_probe --target=device ${grubdir}` || exit 1
 
 # Create the core image. First, auto-detect the filesystem module.
 fs_module=`$grub_probe --target=fs --device ${grub_device}`
@@ -293,9 +293,9 @@ if [ "x${devabstraction_module}" = "x" ]
     if echo "${install_device}" | grep -qx "(.*)" ; then
       install_drive="${install_device}"
     else
-      install_drive="`$grub_probe --target=drive --device ${install_device}`"
+      install_drive="`$grub_probe --target=drive --device ${install_device}`" || exit 1
     fi
-    grub_drive="`$grub_probe --target=drive --device ${grub_device}`"
+    grub_drive="`$grub_probe --target=drive --device ${grub_device}`" || exit 1
 
     # Strip partition number
     install_drive="`echo ${install_drive} | sed -e s/,[0-9]*[a-z]*//g`"
@@ -323,7 +323,7 @@ if [ "x${devabstraction_module}" = "x" ]
         modules="$modules search_fs_uuid"
     fi
 else
-    prefix_drive=`$grub_probe --target=drive --device ${grub_device}`
+    prefix_drive=`$grub_probe --target=drive --device ${grub_device}` || exit 1
 fi
 
 if [ "${target_cpu}-${platform}" = "i386-pc" ] || [ "${target_cpu}-${platform}" = "sparc64-ieee1275" ] ; then
